runtime/codert_vm/cnathelp.cpp

```diff
@@ -970,7 +970,7 @@ old_slow_jitStoreFlattenableArrayElement(J9VMThread *currentThread)
 		} else {
 			if (false == VM_VMHelpers::objectArrayStoreAllowed(currentThread, arrayref, value)) {
 				addr = setCurrentExceptionFromJIT(currentThread, J9VMCONSTANTPOOL_JAVALANGARRAYSTOREEXCEPTION, NULL);
-			} else {
+			} else if ((J9_IS_J9CLASS_VALUETYPE(value)) && (NULL == value)) {
```

Comment:
my mistake, this was fine. Slow path is only a failure path since storing the element cannot fail